staging.kernelci.org

```diff
@@ -245,10 +245,10 @@ cmd_docker() {
     ./kci docker $args kernelci
     ./kci docker $args k8s kernelci
 
-    ./kci docker $args api --version="$api_rev"
+    ./kci docker $args api kernelci --version="$api_rev"
     # TODO - add this functionality to kci docker
-    docker tag kernelci/staging-api:$api_rev kernelci/staging-api:latest
-    docker push kernelci/staging-api:latest
+    docker tag kernelci/staging-api:$api_rev kernelci/staging-api:kernelci
```

Comment:
```suggestion
    docker tag kernelci/staging-api:kernelci-$api_rev kernelci/staging-api:kernelci
```

Retagging built API image was intended to provide a bit more convenient image name (like you did in kernelci/kernelci-api#511).

Resulting tag includes both `kernelci` fragment and API revision ([`build()`](https://github.com/kernelci/kernelci-core/blob/7d62f42b59eefa27ff8821d400ffcc202529bd60/kernelci/cli/docker.py#L97) -> [`get_image_name()`](https://github.com/kernelci/kernelci-core/blob/7d62f42b59eefa27ff8821d400ffcc202529bd60/kernelci/docker.py#L63))

Reply:
Yes, that's why I added `kernelci` tag to the target image tag.
Your suggested change alters the source image tag instead.

See the command help:
`docker tag SOURCE_IMAGE[:TAG] TARGET_IMAGE[:TAG]`

Reply:
Right, because source Docker container image tag changed after adding `kernelci` fragment back.

I might be missing something obvious - please have a look at the steps I did:

1. I checked out your `core` patch which separates core/API fragments ("14ec54cb config/docker/fragment/api.jinja2: use `kernelci` fragment")
2. I set up envs like L232+L239-242 (I ignored disabling cache with `cache_arg`):

```
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ echo $api_rev 
c6c9f63ebf15cac808388d6226075c2b4843380c
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ echo $core_rev 
ede828339b35150ea7e5a7122b69437cd4a8f5f8
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ echo $args
build --prefix=kernelci/staging- --build-arg core_rev=ede828339b35150ea7e5a7122b69437cd4a8f5f8 --build-arg api_rev=c6c9f63ebf15cac808388d6226075c2b4843380c
```

3. I built the new image:

```
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ ./kci docker $args api kernelci --version="$api_rev"
Building kernelci/staging-api:kernelci-c6c9f63ebf15cac808388d6226075c2b4843380c
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ docker image list
REPOSITORY                  TAG                                                 IMAGE ID       CREATED          SIZE
kernelci/staging-api        kernelci-c6c9f63ebf15cac808388d6226075c2b4843380c   11735000d836   3 minutes ago    1.31GB
(snip)
```

4. I attempted retagging like in your PR which didn't work for me:

```
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ docker tag kernelci/staging-api:$api_rev kernelci/staging-api:kernelci
Error response from daemon: No such image: kernelci/staging-api:c6c9f63ebf15cac808388d6226075c2b4843380c
```

5. Then I applied my suggestion from review which retagged the built Docker container image to `kernelci/staging-api:kernelci`:

```
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ docker tag kernelci/staging-api:kernelci-$api_rev kernelci/staging-api:kernelci
(venv-kci-core) pawiecz@c01l:~/c/kci/kernelci-core$ docker image list
REPOSITORY                  TAG                                                 IMAGE ID       CREATED          SIZE
kernelci/staging-api        kernelci                                            11735000d836   4 minutes ago    1.31GB
kernelci/staging-api        kernelci-c6c9f63ebf15cac808388d6226075c2b4843380c   11735000d836   4 minutes ago    1.31GB
```

Please let me know if you see any step that I missed

Reply:
Yes, that was a bug. Thanks for testing the script and proposing the fix.

Reply:
I have updated the scripts now to reflect changes from https://github.com/kernelci/kernelci-core/pull/2495.